.drone.yml

```diff
@@ -0,0 +1,195 @@
+    when:
+      event:
+        - push
+        - pull_request
+        - tag
+
+  - name: test
+    image: docker:19.03
+    volumes:
+      - name: dockersock
+        path: /var/run
+    commands:
+      - apk add git bash curl sudo jq make
+      - sleep 5 # give docker enough time to start
+      - make e2e
+    when:
+      event:
+        - push
+        - pull_request
+        - tag
+
+  - name: build
+    image: golang:1.14
+    commands:
+      - make ci-setup
+      - make build-cross
+    depends_on:
+      - lint
+      - test
+    when:
+      branch:
+        - master
+      event:
+        - push
+        - tag
+
+  - name: release
+    image: plugins/github-release
+    settings:
+      api_key:
+        from_secret: github_token
+      files:
+        - _dist/*
+      checksum:
+        - sha256
+    depends_on:
+      - lint
+      - test
+      - build
+    when:
+      event:
+        - tag
+      refs:
+        - refs/tags/v*
+
+services:
+  # Starting the docker service to be used by dind
+  - name: docker
+    image: docker:19.03-dind
+    privileged: true
+    volumes:
+      - name: dockersock
+        path: /var/run
+
+volumes:
+  - name: dockersock
+    temp: {}
+
+---
+
+kind: pipeline
+type: docker
+name: docs
+
+platform:
+  os: linux
+  arch: amd64
+
+steps:
+
+  - name: build
+    image: python:3.8
+    commands:
+      - python3 -m pip install -r docs/requirements.txt
+      - mkdocs build --verbose --clean --strict
+    when:
+      branch:
+        - master
+      event:
+        - push
+
+  - name: publish
+    image: plugins/gh-pages
+    settings:
+      password:
+        from_secret: github_token
+      username:
+        from_secret: github_username
+      pages_directory: site/
+      target_branch: gh-pages
+    when:
+      branch:
+        - master
+      event:
+        - push
+
+trigger:
+  event:
+    - push
+  ref:
+    - master
+
+---
+
+kind: pipeline
+type: docker
+name: proxy
+
+platform:
+  os: linux
+  arch: amd64
+
+steps:
+  # TODO: use tag from git (without 'proxy-' prefix)
+  - name: build_push
+    image: plugins/docker
+    settings:
+      repo: rancher/k3d-proxy
+      dockerfile: proxy/Dockerfile
+      context: proxy/
+      username:
+        from_secret: dockerhub_username
+      password:
+        from_secret: dockerhub_password
+    when:
+      event:
+        - tag
+      ref:
+        - refs/tags/proxy-v*
+
+trigger:
+  event:
+    - tag
+  ref:
+    - refs/tags/proxy-v*
+
+---
+
+kind: pipeline
+type: docker
+name: tools
+
+platform:
+  os: linux
+  arch: amd64
+
+steps:
+  # TODO: use tag from git (without 'tools-' prefix)
+  - name: build_push
+    image: plugins/docker
+    settings:
+      repo: rancher/k3d-tools
+      dockerfile: tools/Dockerfile
+      context: tools/
+      username:
+        from_secret: dockerhub_username
+      password:
+        from_secret: dockerhub_password
+    when:
+      event:
+        - tag
+      ref:
+        - refs/tags/tools-v*
```

Comment:
you do say you want todo something different here. making new images is free even if they're the same image it should layer cache and not hurt anyone's feelings. maybe you should just pin tools and proxy versions to the cli version for consistency?

Reply:
Yeah, I guess that makes sense! So you would always also build tools and proxy and just release them as new versions even if nothing changed?

Reply:
removed the ref filters